src/coreclr/vm/instmethhash.cpp

```diff
@@ -96,30 +96,19 @@ static DWORD Hash(TypeHandle declaringType, mdMethodDef token, Instantiation ins
 
     DWORD dwHash = 0x87654321;
 #define INST_HASH_ADD(_value) dwHash = ((dwHash << 5) + dwHash) ^ (_value)
+#ifdef TARGET_64BIT
+#define INST_HASH_ADDPOINTER(_value) INST_HASH_ADD((uint32_t)(uintptr_t)_value); INST_HASH_ADD((uint32_t)(((uintptr_t)_value) >> 32))
+#else
+#define INST_HASH_ADDPOINTER(_value) INST_HASH_ADD((uint32_t)(uintptr_t)_value);
+#endif
 
-    INST_HASH_ADD(declaringType.GetCl());
+    INST_HASH_ADDPOINTER(declaringType.AsPtr());
     INST_HASH_ADD(token);
```

Comment:
```suggestion
```

I think we can remove including the `token` now as well. If we leave this, I think we need to add a comment as to why the pointer hashing isn't enough.

Reply:
The token is to identify twhich method on the type, so we still need that

Reply:
Oh right. That makes sense as the pointer is just for the type and not the member. Okay. Looks good to me.